### DISCUSSION TITLE
Project: Multilayer_SMA2Microstrip

### DISCUSSION THREAD

**Discussion by toammann** [thliebig/openEMS-Project#88] [Show and tell]
Hello,
I have just finished my SMA coaxial to microstrip transistion project using open source software only (mainly openEMS). 

The final result is an easy-to-use KiCAD footprint. The footprint was designed for the Rosenberger 32K242-40ML5 SMA connector on an low-cost AISLER 6-layer FR4 stackup:

This is my first project using openEMS. I am very surprised how well this solver works. The report also contains a VNA measurement. I think it is very important to publibsh measured results side by side with simulated ones. This will greatly improve the confidence in the simulator and hopefully attract more users - which in turn results in more documentation especially for beginners.

The boards were fabricated by AISLER a German rapid PCB prototyping service. Starting from 12/2022 they have released a new FR4 6-layer PCB stackup that should works well for RF projects. One very nice feature of AISLER is that "Castellated Holes" are free. I had the idea to abuse this castellated hole feature to create a short piece of edge metallization.

![connector_openEMS](https://github.com/thliebig/openEMS-Project/assets/9382233/e6356f5d-94f1-4757-9707-bec350ef8904)

Please find the project documentation here:
[GitHub Multilayer_SMA2Microstrip](https://github.com/toammann/Multilayer_SMA2Microstrip)

**Comment by PinkMushroom** [comment]
Really excellent write up !
It's really nice to see your use of free software for this project (I have not had particularly good luck using Freecad so I'm looking forward to importing your model and seeing how it works).

> I think it is very important to publibsh measured results side by side with simulated ones

It really is.  

I think this is a really nice test case since one of the drawback to FDTD is the staircasing on curved sections.

Even with the subgridding I would think that the gird would have to be extremely fine for this problem, it's a good example of a situation where FE methods might be more efficent.

- How many cells were used to model the problem space ?
- Do you have any simulation time data ?

I noticed that S21 in your simulations does not have a slope.

- Does OpenEMS model all metal as PEC ?
- Did you use a constant value for the permittivity of the PCB material ?

Thanks very much for taking the time to publish these results !

**Reply by toammann** [reply]
Hello PinkMushroom,

> Even with the subgridding I would think that the gird would have to be extremely fine for this problem, it's a good example of a situation where FE methods might be more efficent.

I think the error is not so big as one might think. I don't know how the structure is discretized exactly. Maybe someone can give some insight here? By activating the PEC dump, we can take a look at the discretized PEC structure. 

![bitmap](https://github.com/thliebig/openEMS-Project/assets/9382233/b64ea385-b593-4343-b667-ace5b23e8b33)

Here we can see, that at some points of the circle's  diameters will look larger but at other points they appear smaller. So, overall, some "averaging" effect will come into play. The exact shape of the coax does not matter so much for the impedance as long as the inner to outer distance (which defines) the capacitance is constant (at least to my knowledge). I have run a simulation with a more coarse mesh(mesh_div=40) and the result is nearly the same.

Commercial solvers like CST Microwave studio have proprietary techniques to solve this issue in the time domain mesh ([Perfect Boundary Approximation](https://www.researchgate.net/file.PostFileLoader.html?id=578c450ceeae3937441b63a1&assetKey=AS%3A385033333428224%401468810508239))

The simulation space has about 12e6 mesh cells. On my computer, the excitation of one port takes about 20 min.
```
FDTD simulation size: 282x267x157 --> 1.18212e+07 FDTD cells
FDTD timestep is: 3.7023e-14 s; Nyquist rate: 450 timesteps @3.00114e+10 Hz
```

I think that the time step has a larger influence on the simulation duration than the number of mesh cells. I have to discretize the very thin edge plating (20um). This should define my smallest mesh cell and make my time step small. 

> Does OpenEMS model all metal as PEC ?

No, the metal layers are modeled as Copper sheets with their physical thicknesses (35um). However, the thickness is discretized by one mesh cell only. There is also no copper roughness model applied.

Does anyone know if the skin effect is considered is this case? The currents are solved on the edges of the meshcells. So will the current look like it flows at the edges anyway?

> Did you use a constant value for the permittivity of the PCB material ?

Yes, the permittivity and also tand are constant.

Please feel free to correct me if something I have written is not correct.

Regards!

**Reply by VolkerMuehlhaus** [reply]
I'm wondering if you really got constant loss tangent in your model. Or more general, if dielectric loss can be defined correctly in openEMS for broad band models.

To my understanding, openEMS doesn't support loss tangent and instead uses conductivity for material objects. In your code, we can see how tand is converted to the equivalent **conductivity at center frequency**. However, that conversion is valid at that exact frequency, and not for broad band simulation !?

Snippet from your code:
`mat_core.tand = 0.015;
mat_core.kappa = mat_core.tand*mat_core.er*EPS0*2*pi*(f_stop-f_start)/2;
(...)
CSX = SetMaterialProperty( CSX, mat_core.name, 'Epsilon', mat_core.er, 'Kappa', mat_core.kappa);`

The issue is that constant conductivity (openEMS parameter kappa, otherwise called sigma) is not constant loss tangent !? Am I missing something?

![grafik](https://github.com/thliebig/openEMS-Project/assets/47922981/e1ecca25-60c6-4f72-a7c9-5583c2ec6ec3)

**Reply by toammann** [reply]
I think you are right here. It is incorrect to say "The simulation was performed at a constant tand"

By using the equation:
```
mat_core.tand*mat_core.er*EPS0*2*pi*(f_stop-f_start)/2
```
I set the dielectric damping epsilon'' to zero assuming that the loss is only caused by the conductivity of the material. 

The conductivty sigma is a frequency independet value.  So, the tand and permittivty values used to calculate sigma must be valid at the chosen frequency (here mid band). 

I assume the correct statement should be:  "The simulation was performed at a constant kappa". 

Does this also mean, that with a constant permittivity and Kappa of a material the tand is proportional to 1/f?  Will the loss tangent decrease towards zero for high frequencies?  Is this right? :)

This may excplain why the flat S21 curve does not show a loss increase for higher frequencies.

I am defenitly an expert on this topic. So opinions are appreciated :)

**Comment by PinkMushroom** [comment]
> I think the error is not so big as one might think. I don't know how the structure is discretized exactly. Maybe someone can give some insight here? By activating the PEC dump, we can take a look at the discretized PEC structure.

yes, the issue is the number of cells that you need to get the desired accuracy.  I have not seen good comparisons of FDTD vs FEM, and I would suspect that it's problem dependent.  However for something with a lot of arcs/curves you would think that FE would be more efficient.

> No, the metal layers are modeled as Copper sheets with their physical thicknesses (35um). However, the thickness is discretized by one mesh cell only. There is also no copper roughness model applied.

The metal surface is probably being treated as PEC, and so there is not skin depth loss.  This something to look into and could be a nice OpenEMS project for me to add it ;-)

> Please feel free to correct me if something I have written is not correct.
I think somebody who has more experience with OpenEMS can comment on that.

Please check your Multilayer_SMA2Microstrip github project. I have opened two issues.  I can't get the project to run :-(

**Comment by PinkMushroom** [comment]
I was successful in making the project run after what basically were very minor issues

1 the necessary paths for octave were not set
/opt/openEMS/share/CSXCAD/matlab
/opt/openEMS/share/openEMS/matlab

2 there were some problems with the file paths using "\\" instead of "/"

```
Time for 18480 iterations with 11821158.00 cells : 965.62 sec
Speed: 226.23 MCells/s 
error: 'read_touchstone' undefined near line 807, column 807
error: called from
    aisler_sma_v2 at line 807 column 18
```
And still one minor problem in that read_touchstone appears to be missing.  I did find it in openEMS/CTB, I just need to copy it over or add another octave path variable.

Only 16 minutes ! I think that is quite good for a project of this complexity.  I have run similar projects in other EM tools and they have tended to run _slower_ than this , even with parallelization across many cores.

**Reply by toammann** [reply]
Hey,
glad to hear that it works for you now. I have tested the project also in linux. It runs now without a problem.

I noticed that the simulation is about 20% faster in Linux compared to Windows. 

```
- Windows:   approx. 18.5min
- Linux:     approx. 14.75min 
```

The simulations run on the same machine which is set up in a dual boot configuration. Does anyone know the reason for this?

**Reply by biergaizi** [reply]
openEMS's current multi-threaded engine makes heavy use of thread synchronization, during a single timestep there can be ~20 synchronization steps, much of CPU time of the threads is used on just waiting for the barriers to unlock. So my educated guess is that even a slight difference of OS's CPU task scheduler can make difference in runtime. My experience shows the engine can also be pretty sensitive to CPU clock frequency, even in a supposedly memory-bound task, likely due to this locking issue.

**Comment by biergaizi** [comment]
Using my experimental diamond tiling engine (currently under development), this project achieved a 400% speedup.

**Reply by toammann** [reply]
Hey, this sounds really amazing! I still did not find the 
time to read your thread carefully :(

With a 400% speedup the simulation duration would be, in my experience,   comparable to CST for projects of this kind  (Maybe even faster?).

**Comment by biergaizi** [comment]
> Did you use a constant value for the permittivity of the PCB material?

openEMS does not have out-of-the-box support for materials with frequency-dependent permittivity. But openEMS includes model for Drude and Lorentz materials, and they can be used to model frequency-dependent PCB material.

See:

* https://wiki.openems.de/index.php/Drude/Lorentz_Material_Property.html
* https://empossible.net/wp-content/uploads/2018/03/Lecture-2-Lorentz-and-Drude-models.pdf

The workflow is basically:

1. Measure the frequency-dependent complex permittivity of a real PCB material like FR-4 (the permittivity you get from most PCB manufacturers is basically a lie, because there's no single right number, it must be actually measured in the frequency domain).
2. Fit the parameters of the Lorentz material to the PCB by numerical optimization, by developing your own tools.
3. Create the Lorentz material with the parameters you found in openEMS.

Proprietary field solvers often include a database of common materials. I found one field solver claims they have the largest database because they went through the trouble of fitting a lot of materials. 

I do not know if an open database exists.

**Reply by 0xCoto** [reply]
> Fit the parameters of the Lorentz material to the PCB by numerical optimization, by developing your own tools.

I haven't looked too much into dispersive materials so I'm not an expert on that type of optimization in particular, but from a quick look [PhySo](https://github.com/WassimTenachi/PhySO) may be usable as part of a fun experiment. Though I'm not sure how intelligent the optimizer has to be as each evaluation appears computationally inexpensive? Might look into this eventually.

> I found one field solver claims they have the largest database because they went through the trouble of fitting a lot of materials.

From a quick search that seems to be claimed by COMSOL? It's unclear whether that applies to their general physics solvers though or specifically RF simulation package. I'd also doubt they're using frequency-dependent models for their default material library (though I can check) - from what I've seen, proprietary solvers have a material library of common materials with a constant dielectric constant (which is hopefully "close enough" for most frequencies), but they do allow you to specify dispersive materials given a set of frequency and $ε_r$ values you input.

Perhaps they're running a quick optimization like the one you mentioned - haven't checked.

**Reply by biergaizi** [reply]
I found this Simberian presentation that gives a good overview on modeling dispersive materials for PCB design:

* [Modeling frequency-dependent dielectric loss and dispersion for multi-gigabit data channels](https://www.simberian.com/AppNotes/ModelingDielectrics_2008_06.pdf)

Apparently the Djordjevic-Sarkar model is common for electronics design. Its robustness has been confirmed in multiple IEEE and DesignCon papers. The best advantage is that it allows the use of dielectric constant and loss tangent as few as a single frequency point only, while still gives reasonable results. So to model PCBs, one can use this model to generate the complex permittivity curve, then we can fit a multi-pole Drude/Lorentz model based on the curve.

This should be the route of investigation if anyone wants to solve the PCB modeling problem.

And speaking of metal modeling and skin depths, I found [XFDTD's (XF) documentation](https://support.remcom.com/xfdtd/reference/definitions/surface-conductivity-correction.html) provides a useful description of the problem:

> The finite-difference time-domain (FDTD) method requires a minimum of ten cells per wavelength to assume a linear field transition between adjacent cells in order to consistently achieve accuracy. Maintaining this relationship for good conductors is not practical because the fields are decaying exponentially over the skin depth, which is a short distance when compared to the free space wavelength. The surface conductivity correction in XF can be enabled for good conductors to more accurately model the losses and reflections at the boundary of a metal.

So it's not practical to directly model skin effect using FDTD. Modeling the metal thickness means modeling an extremely thin layer of conductor and the grid size required would be too big to be practical (when trying to model the solder mask, I've encountered the same problem). The best we can do is a lossy correction. This answers the question from @PinkMushroom 

Find a suitable correction method for copper surface roughness would be important for high-speed digital design in the 20-100 Gbps realm.

**Reply by toammann** [reply]
Thanks for sharing this information.

@biergaizi 
> 1. Measure the frequency-dependent complex permittivity of a real PCB material like FR-4 (the permittivity you get from most PCB manufacturers is basically a lie, because there's no single right number, it must be actually measured in the frequency domain).
  >2. Fit the parameters of the Lorentz material to the PCB by numerical optimization, by developing your own tools.
  >3. Create the Lorentz material with the parameters you found in openEMS.

I totally agree that this is the "ideal" workflow. However, it is unpractical for most people. How many people have access to a measurement setup to measure the permittivity? Big (RF) companies have people who are committed to this task (build a company internal materiallibrary for accurate simulations). But this is expensive. 

So, in many cases you have to rely on the manufacturer data of the substrate materials. Which must - as you pointed out - be treated with care. This is actually a huge benefit of the PCB manufacturer I have chosen here. They define a layer stack which is always the same (this is NOT true for every PCB-Prototyping service, Substrate materials and thicknesses may vary). Aisler uses the Panasionic R-1566(W)/R-1551(W) FR4 material.  Panasonic provides frequency dependent values for permittivity and DK. 

Even though the permittivity variation is not simulated one can estimate the influence on the simulation result by having this data available. The permittivity change es from 4.3@1GHz to 4.2@10GHz. This variation is minor for this use case. For lower frequencies the permittivity increases but the electrical length of the structure decreases. 

What I do not understand about the simulation result is the following: 
Even though permittivity and tanD are constant in the material definition the electrical length of the structure is not. It increases with frequency. So higher frequencies should be attenuated more just by the fact that the wave has "a longer way" in the model. At DC I would expect the insertion loss to hit 0dB because the electrical length goes towards 0.

The simulation results do not behave like this. They are totally flat with a non-zero insertion loss at DC.

**Reply by biergaizi** [reply]
> I totally agree that this is the "ideal" workflow. However, it is unpractical for most people. How many people have access to a measurement setup to measure the permittivity? 

I already proposed a solution without the need for measurement - Djordjevic-Sarkar model.

**Reply by biergaizi** [reply]
> At DC I would expect the insertion loss to hit 0dB because the electrical length goes towards 0. The simulation results do not behave like this. They are totally flat with a non-zero insertion loss at DC.

If my understanding is correct, FDTD, at least in its unmodified form, is inherently unsuitable for DC. To get accurate low frequency results, one needs an extremely long excitation signal with timesteps that are many orders of magnitude higher than what can be practically simulated on a computer. Even if it can be done, numerical stability problems can cause low-frequency breakdown. 

So I consider the near-DC result of any FDTD simulators to be invalid without giving it a second thought. This is not a serious limitation, since Vector Network Analyzers don't work at DC either. Using [DC extrapolation](https://scikit-rf.readthedocs.io/en/latest/examples/networktheory/DC%20Extrapolation%20for%20Time%20Domain%20.html) should suffice for most purposes.

Perhaps @thliebig can help clarify the problem and comment on whether my understanding is correct.

**Comment by VolkerMuehlhaus** [comment]
> So it's not practical to directly model skin effect using FDTD. 

One solution used in commercial FDTD is to model hollow thick metal conductors, with zero thickness surface sheets all around. Surface impedance of these sheets then includes skin effect, similar to the openEMS conducting sheet model.

In openEMS, I actually modelled skin effect in RFIC transmission lines (where all dimensions are rather small) by meshing into skin depth with sub-micron mesh density. This was required because these lines can't be considered flat, thickness matters for the actual EM solution. Results are quite good, but this simulation is slow due to small timestep and many cells ... not a solution for PCB.

![line1080_s21](https://github.com/thliebig/openEMS-Project/assets/47922981/7f221856-20ef-40ae-b78a-6b4970d6afe2)

Regarding RF losses in PCB lines, my experience is that it's dielectric loss that matters most. We really need to include loss tangent in these cases, it has much more effect that conductor loss. Those expensive low loss RF substrates have the same copper as normal PCB, but very low loss dielectrics.

**Reply by biergaizi** [reply]
> Regarding RF losses in PCB lines, my experience is that it's dielectric loss that matters most. We really need to include loss tangent in these cases, it has much more effect that conductor loss. 

Of course, dielectric loss is the predominant factor, conductor loss is unimportant. But if we're already going through the trouble of modeling dispersive microstrip, correcting for surface roughness in the future would be a nice bonus...

**Reply by PinkMushroom** [reply]
> conductor loss is unimportant

Not true, as you might expect it depends on operating frequency.

At high frequencies conductor loss is very important and dielectric loss less so.

There is a good reason for this. You must use very low loss dielectrics at high frequencies, and so the conductor loss dominates. Good RF materials will have a loss tangent of <= 0.0015.

Above 15-20GHz, surface roughness is _required_ or your loss calculation will be too low. Not a problem if your trace is 1cm long, a big problem if it's 10cm long.

Some EM programs handle it as an a property attached to the conductor faces.

Other programs attach it as a property of the material which is , strictly speaking, incorrect. It is well known that the roughness differs depending on which side of the copper you are on.  Also, it's not a material property but a "mechanical" property.

Even more interesting is that the SR will affect the impedance. There is some interesting information from Rogers on that subject.

As we all know handling frequency dependent effects in the time domain is a headache.  However once it can be done, I assume using recursive convolution, it allows us to handle both SR, skin depth and frequency dependent dielectric properties uniformly (I think).

**Reply by biergaizi** [reply]
> At high frequencies conductor loss is very important and dielectric loss less so.

Everything is in relative terms, depending on the applications. I first claimed that "Find a suitable correction method for copper surface roughness would be important for high-speed digital design in the 20-100 Gbps realm" , so certainly I didn't think it's totally unimportant. Then I later said "conductor loss is unimportant" in agreement to MWit83, because we don't have a accurate model for dielectric loss of the substrate for now (and I mentioned a potential way to achieve that) so it's a problem with a higher priority.

And if we have already reached the point of considerations surface roughness, another two interesting effects also worth consideration: ENIG plating loss and substrate anisotropy - they would also be a nice bonus to model in the future...

> As we all know handling frequency dependent effects in the time domain is a headache.

Handing frequency-dependent effects in time domain is a "complex" and "convoluted" problem ;-)

**Reply by toammann** [reply]
Just for the sake of completeness:
CST Microwave Studio has newly implemented the "[Gradient Model](https://ieeexplore.ieee.org/document/6483879)". It requires only one input Parameter: The surface roughness.

I have not tried how accurate you can get with this. The problemen here is again: How to get a reliable number for the surface roughness. From a PCB manufacturer you may only geht the maximum allowed value and not the "typical" value you need to get a good agreement between measured and simulated results.

Maybe a "practical" approach is here to build a small piece of transmission line and tune the surface roughness parameter until the simulation measurement results agree (assumed the dielectric losses are modelled correctly)? 

Also: The conductor loss  is significantly influenced ty the trace width. Rogers has published a lot of materials about this topics. 

![grafik](https://github.com/thliebig/openEMS-Project/assets/9382233/154315c4-0a7b-4753-ac15-f5bd85023409)
[Circuit Materials and High-Frequency Losses of PCBs](https://www.rogerscorp.com/-/media/project/rogerscorp/documents/articles/english/advanced-connectivity-solutions/circuit-materials-and-high-frequency-losses-of-pcbs.pdf)

In this data the conductor loss is significant. It is even higher than the dielectric loss.

I think they even have a calculator that splits conductor and dielectric losses for their products. I have not looked into this.
https://rogerscorp.com/blog/2018/calculate-microwave-impedance-with-transmission-line-modeling-tool

**Comment by PinkMushroom** [comment]
> And if we have already reached the point of considerations surface roughness, another two interesting effects also worth consideration: ENIG plating loss and substrate anisotropy - they would also be a nice bonus to model in the future...

All of these are variations on a similar problem, simulating thin metals where the grid density would be too small.
You would think there would be a lot of research on the subject. Time to exercise ieeexplore ;-)


> Handing frequency-dependent effects in time domain is a "complex" and "convoluted" problem ;-)

LOL.

**Comment by VolkerMuehlhaus** [comment]
> All of these are variations on a similar problem, simulating thin metals where the grid density would be too small. 

For the testcase shown here, I would use openEMS thin metal model with built-in skin loss (AddConductingSheet) instead of generic material (AddMaterial). 

There is no relevant side wall coupling, and the benefit of including physical metal thickness in the actual field solution is very small here. The thin sheet model (with surface impedance including skin effect) should work well here, it captures skin effect and and there is no trouble with small mesh cells.

AddConductingSheet example is included for Matlab: openEMS\matlab\examples\transmission_lines\MSL_Losses.m

**Reply by biergaizi** [reply]
> I would use openEMS thin metal model with built-in skin loss (AddConductingSheet) instead of generic material (AddMaterial).

I'm aware of the existence of `AddConductingSheet`, but I ignored it because it's useful but technically uninteresting (if you read the code, you would see it's basically just a Lorentz material with pre-defined parameters, found by numerical optimization using the script [OptimizeCondSheetParameter.m](https://github.com/thliebig/openEMS/blob/master/FDTD/extensions/OptimizeCondSheetParameter.m)). If the requirement changes slightly, new approximations are needed.

This is why I started discussing on alternative methods of modeling, including whether skin effect could be modeled directly in 3D FDTD alone [1]. I'm glad to hear the result from @PinkMushroom that it's indeed possible to directly simulate skin effect directly without making any assumption - though it's impractical for most practical applications due to the need of an unrealistically fine mesh.

So the conclusion is mostly negative, it's mostly an impractical and futile attempt - keep constructing more accurate material model is the practical solution...

[1] One of my motivations is seeing whether it's possible to model the effect of solder-mask coated transmission lines - this is a standard task for 2D frequency-domain solvers and they're used every day in electronics design. Comparing the result of 2D MoM solvers and 3D FDTD solvers would be interesting from a technical perspective - since 3D full-wave solvers are generally believed to be more powerful since they work from "first principles" (certainly, as we have realized, it's not the case for FDTD when modeling extremely thin structures...) This problem can be practically important too - according to 2D MoM simulations, in a differential coplanar waveguide, even a thin solder mask filling between the traces can significant affect its characteristic impedance since the solder mask acts as a dielectric material. This effect is not possible to model if the metal is zero-thickness. To study this effect, the conclusion appears to be that, since we already know that the skin effect can be directly simulated with an extremely fine mesh, this must be equally true for a layer of thin coating above the metal. And for the purpose of technical investigation, this test is possible. On the other hand, for practical problems, the solution is probably an approximation similar to `AddConductingSheet`.

**Reply by toammann** [reply]
@MWit83 
> For the testcase shown here, I would use openEMS thin metal model with built-in skin loss (AddConductingSheet) instead of generic material (AddMaterial).

One could of course do this. However this would break the modeling workflow I have chosen for this task. The idea was to model everything in FreeCAD including the conductors with the small matching network. I think the project is easier to maintain when everything everything is modeled in one place (either FreeCAD or openEMS). 

For a good RF transition it is key to optimize the return loss. So, it is imporant to get this right. Another point of consideration is to prevent radiation which is visible in the insertion loss. These effects can be observed even though the conductor losses are not precisely modeled. Predicting the insertion loss was not my focus in this project.

**Reply by VolkerMuehlhaus** [reply]
Yes, I agree from a workflow perspective.

However, from my 20+ years as full time EM support specialist with commercial solvers, I recommend to include proper conductor loss (with skin effect) and that is missing if you go for thick metal model using AddMaterial. 

It's a tradeoff between error from loss (skin effect) and error from conductor height. Here with these geometries, conductor height has almost no effect, so I would recommend to go for the more accurate insertion loss by including skin effect -> lossy thin sheet. 
For other cases where side wall capacitance matters, thick metal model can be more important.

~~

Regardless of that detail: excellent work, congratulations!

**Comment by PinkMushroom** [comment]
>  though it's impractical for most practical applications due to the need of an unrealistically fine mesh.

I've been doing a preliminary search, and it looks like this is handled using what's called a surface impedance boundary condition. The technique is, not surprisingly, quite old.  It appears to fold into the FD scheme cleanly, but i haven't spent too much time on it , so I could be completely wrong.

I'm also fairly certain that I've seen someone else in another discussion thread mention that technique also.

> One of my motivations is seeing whether it's possible to model the effect of solder-mask coated transmission lines

That is indeed an underrated problem. Especially significant for high speed digital where the substrate may only 100um and the soldermask 25u.  The soldermask has (can have) a high dielectric constant and is definitely not low loss. LOL.

Once again this all points to why FE has, in many cases, taken over the EM simulation space. Sure you end up with a very fine mesh in those areas, but it happens "automagically" and you can simply enter material parameters and be done with it. Then you throw 40 or 50 cores at your problem and don't worry about it.  In the commercial space this is what happens.  However those extra cores cost money (the software charges for them !!), so there is an advantage to having efficient handling of these situations whenever possible.

However I'm having doubts about whether that's really true. This problem, for it's complexity, and the fine mesh that it uses, simulates quite fast.  Remember I only have 8 cores !  I have absolutely simulated problems of similar complexity on "fast machines" with commercial software and they have taken as long or _longer_.

**Reply by toammann** [reply]
Hey PinkMushroom,
I kindly ask you to write a reply to the comment instead of opening  a new comment. The thread gets a bit cluttered :) You may also copy your comment as a reply to the suitable comment :)

**Comment by null-a** [comment]
@toammann Lovely stuff! I found this while trying to teach myself something about this topic (as a hobby). Reading your work is enlightening, so thanks for sharing!

One thing I'm curious about is the capacitive compensation structure, and in particular how you arrived at it? Perhaps a discontinuity and its location was discovered by TDR? And I wonder if you could say something about the structure itself? I don't know what I'm looking at, though it looks kinda like a tiny stepped impedance filter or something. Any pointers for a novice would be appreciated! Thanks again!

**Reply by toammann** [reply]
Hello null-a,
Happy to hear that you are interested in the topic. To answer your questions:

The compensation structure is there to "fine-tune" the matching. I think of it as an LC-Matching element.

The design strategy I chose for the connector was to minimize the parasitics of the transition as much as possible before the design of the compensation structure. Of course there are other stradegies to this (especially for dieleictrics with a lower Er). So, I used the openEMS model to verify that no radiation is taking place and the GND return current inductance is minimal. It is also important to minimize the pad capacitance or to have it close to the line impedance of 50Ohm.

At some place, I have settled to this design stage and exported a touchstone file from my openEMS simulation. At this point, the reference plane was right at the end of the pad (there was no compensation structure). 

The design of the compensation structure is then very easy and can be done graphically. If I take a look at the reflection coefficient in QUCS at the end of the pad (having the reference plane at the right position is key!)  it is clear that the matching is reasonable already. However, the trajectory is shifted slightly into the capacitive lower right area. This is mostly due to the return current inductance combined with the phase shift of the pad (think of it as a transmission line).

![grafik](https://github.com/thliebig/openEMS-Project/assets/9382233/1454caa7-b4b9-4dc6-b970-39ed0ca0ffa2)

To have the best broadband match, one wants the trajectory to be centered around the origin. This can be accomplished by a simple LC-ladder network. In QUCS studio, we can use the tuner to achieve this goal in seconds. An alternative (more sophisticated) design approach might be to create an equivalent circuit of the touchstone file and absorb this into a stepped impedance low pass filter. However, in this case, using the tuner in QUCS is by far the fastest solution.

![grafik](https://github.com/thliebig/openEMS-Project/assets/9382233/a6f09feb-6618-498b-adff-093c993d0edf)

Now you have the value for of the compensation inductor and capacitor. The last step is to transform the lumped components into an equivalent transmission line implementation ([Richards’s Transformation ](https://eng.libretexts.org/Bookshelves/Electrical_Engineering/Electronics/Microwave_and_RF_Design_IV%3A_Modules_(Steer)/02%3A_Filters/2.12%3A_Richardss_Transformation)). Now you have the physical dimensions of the compensation structure as a starting point for the EM stimulation. 

![grafik](https://github.com/thliebig/openEMS-Project/assets/9382233/30b2fcab-79b2-41a9-9025-88009cda0a2c)

The values taken from QUCS matched the EM simulation very good. So only minor tuning in openEMS was necessary.

Regards,

**Reply by null-a** [reply]
Thanks for taking the time to write this, it's really helpful.

When I first saw the layout I mistook what I think might be called a solder mask dam for another C, but now you point it out I see this is an L&C only. The qucs workflow for tuning the match sounds slick, I'll have to play with this myself.

Thanks again.
